Proxmox VE 9 - Fullsync not possible - Too many requests / database is locked

Hey, 

thank you for closing Issue #417 - The connection works now (after reinstalling proxbox api and plugin - update didn't work, the API Server would not start...)

However I am now able to connect to the proxmox instance (using root credentials to prevent any errors from this side) - but I have 3 problems preventing the sync however I try it...

1. Sometimes I get an error that the database file is locked if if no job is active. not sure if you forget to unlock it or whatever... - A restart of the LXC finxes the problem. For reference the error log:
```
proxbox [2026-05-16 13:00:20,613] [ERROR] exceptions: Unhandled exception on GET /proxmox/endpoints
Traceback (most recent call last):
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 1967, in _exec_single_context
    self.dialect.do_execute(
    ~~~~~~~~~~~~~~~~~~~~~~~^
        cursor, str_statement, effective_parameters, context
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/sqlalchemy/engine/default.py", line 952, in do_execute
    cursor.execute(statement, parameters)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: database is locked

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/starlette/middleware/errors.py", line 164, in __call__
    await self.app(scope, receive, _send)
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 191, in __call__
    with recv_stream, send_stream, collapse_excgroups():
                                   ~~~~~~~~~~~~~~~~~~^^
  File "/usr/lib/python3.13/contextlib.py", line 162, in __exit__
    self.gen.throw(value)
    ~~~~~~~~~~~~~~^^^^^^^
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/starlette/_utils.py", line 87, in collapse_excgroups
    raise exc
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 193, in __call__
    response = await self.dispatch_func(request, call_next)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/proxbox_api/app/factory.py", line 161, in dispatch
    return await call_next(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 168, in call_next
    raise app_exc from app_exc.__cause__ or app_exc.__context__
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 144, in coro
    await self.app(scope, receive_or_disconnect, send_no_error)
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 191, in __call__
    with recv_stream, send_stream, collapse_excgroups():
                                   ~~~~~~~~~~~~~~~~~~^^
  File "/usr/lib/python3.13/contextlib.py", line 162, in __exit__
    self.gen.throw(value)
    ~~~~~~~~~~~~~~^^^^^^^
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/starlette/_utils.py", line 87, in collapse_excgroups
    raise exc
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/starlette/middleware/base.py", line 193, in __call__
    response = await self.dispatch_func(request, call_next)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/proxbox_api/app/factory.py", line 205, in dispatch
    authorized, error_message = check_auth_header_with_session(session, api_key, client_ip)
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/proxbox_api/auth.py", line 118, in check_auth_header_with_session
    if is_locked_out(session, client_ip):
       ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/proxbox_api/auth.py", line 93, in is_locked_out
    return AuthLockout.is_locked_out(session, ip, _MAX_FAILED_ATTEMPTS, _LOCKOUT_DURATION)
           ~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/proxbox_api/database.py", line 168, in is_locked_out
    lockout = session.get(AuthLockout, ip)
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/sqlalchemy/orm/session.py", line 3680, in get
    return self._get_impl(
           ~~~~~~~~~~~~~~^
        entity,
        ^^^^^^^
    ...<7 lines>...
        bind_arguments=bind_arguments,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/sqlalchemy/orm/session.py", line 3859, in _get_impl
    return db_load_fn(
        self,
    ...<5 lines>...
        bind_arguments=bind_arguments,
    )
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/sqlalchemy/orm/loading.py", line 695, in load_on_pk_identity
    session.execute(
    ~~~~~~~~~~~~~~~^
        q,
        ^^
    ...<2 lines>...
        bind_arguments=bind_arguments,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/sqlmodel/orm/session.py", line 136, in execute
    return super().execute(
           ~~~~~~~~~~~~~~~^
        statement,
        ^^^^^^^^^^
    ...<4 lines>...
        _add_event=_add_event,
        ^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/sqlalchemy/orm/session.py", line 2351, in execute
    return self._execute_internal(
           ~~~~~~~~~~~~~~~~~~~~~~^
        statement,
        ^^^^^^^^^^
    ...<4 lines>...
        _add_event=_add_event,
        ^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/sqlalchemy/orm/session.py", line 2249, in _execute_internal
    result: Result[Any] = compile_state_cls.orm_execute_statement(
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        self,
        ^^^^^
    ...<4 lines>...
        conn,
        ^^^^^
    )
    ^
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/sqlalchemy/orm/context.py", line 306, in orm_execute_statement
    result = conn.execute(
        statement, params or {}, execution_options=execution_options
    )
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 1419, in execute
    return meth(
        self,
        distilled_parameters,
        execution_options or NO_OPTIONS,
    )
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/sqlalchemy/sql/elements.py", line 527, in _execute_on_connection
    return connection._execute_clauseelement(
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        self, distilled_params, execution_options
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 1641, in _execute_clauseelement
    ret = self._execute_context(
        dialect,
    ...<8 lines>...
        cache_hit=cache_hit,
    )
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 1846, in _execute_context
    return self._exec_single_context(
           ~~~~~~~~~~~~~~~~~~~~~~~~~^
        dialect, context, statement, parameters
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 1986, in _exec_single_context
    self._handle_dbapi_exception(
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        e, str_statement, effective_parameters, cursor, context
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 2363, in _handle_dbapi_exception
    raise sqlalchemy_exception.with_traceback(exc_info[2]) from e
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/sqlalchemy/engine/base.py", line 1967, in _exec_single_context
    self.dialect.do_execute(
    ~~~~~~~~~~~~~~~~~~~~~~~^
        cursor, str_statement, effective_parameters, context
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/proxbox-api/venv/lib/python3.13/site-packages/sqlalchemy/engine/default.py", line 952, in do_execute
    cursor.execute(statement, parameters)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) database is locked
[SQL: SELECT authlockout.ip_address AS authlockout_ip_address, authlockout.attempts AS authlockout_attempts, authlockout.first_attempt_time AS authlockout_first_attempt_time 
FROM authlockout 
WHERE authlockout.ip_address = ?]
[parameters: ('192.168.30.15',)]
(Background on this error at: https://sqlalche.me/e/20/e3q8)
```
2. It seems that somewhere is a rate limit triggered - not sure where to look if I am honest - it looks like that proxmox has a rate limit enabled but I cannot find the option. I already reduced the number of parallel requests in the settings but I regulary get the followint 429 - Too many requests error and after that the job quits. Maybe you can add a automatic cooldown if a 429 is returned...
Here the error log:
```
INFO:     192.168.30.15:36588 - "GET /proxmox/endpoints HTTP/1.1" 200 OK
INFO:     192.168.30.15:36620 - "GET /dcim/devices/create/stream?use_guest_agent_interface_name=true&fetch_max_concurrency=2&ignore_ipv6_link_local_addresses=true&primary_ip_preference=ipv4&overwrite_device_role=true&overwrite_device_type=true&overwrite_device_tags=true&overwrite_device_status=true&overwrite_device_description=true&overwrite_device_custom_fields=true&overwrite_vm_role=true&overwrite_vm_type=true&overwrite_vm_tags=true&overwrite_vm_description=true&overwrite_vm_custom_fields=true&overwrite_cluster_tags=true&overwrite_cluster_description=true&overwrite_cluster_custom_fields=true&overwrite_node_interface_tags=true&overwrite_node_interface_custom_fields=true&overwrite_storage_tags=true&overwrite_vm_interface_tags=true&overwrite_vm_interface_custom_fields=true&overwrite_ip_status=true&overwrite_ip_tags=true&overwrite_ip_custom_fields=true&overwrite_ip_address_dns_name=true&proxmox_endpoint_ids=4 HTTP/1.1" 429 Too Many Requests
INFO:     192.168.30.15:36626 - "PUT /proxmox/endpoints/1 HTTP/1.1" 429 Too Many Requests
INFO:     192.168.30.15:36638 - "PUT /proxmox/endpoints/1 HTTP/1.1" 429 Too Many Requests
INFO:     192.168.30.15:36646 - "PUT /proxmox/endpoints/1 HTTP/1.1" 429 Too Many Requests
INFO:     192.168.30.15:36678 - "GET /proxmox/endpoints HTTP/1.1" 429 Too Many Requests
INFO:     192.168.30.15:36680 - "GET /proxmox/endpoints HTTP/1.1" 429 Too Many Requests
INFO:     192.168.30.15:36662 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:36700 - "GET /proxmox/endpoints HTTP/1.1" 429 Too Many Requests
INFO:     192.168.30.15:36684 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:36712 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:36722 - "GET /proxmox/endpoints HTTP/1.1" 429 Too Many Requests
INFO:     192.168.30.15:36738 - "GET /proxmox/endpoints HTTP/1.1" 429 Too Many Requests
INFO:     192.168.30.15:36750 - "GET /proxmox/endpoints HTTP/1.1" 429 Too Many Requests
INFO:     192.168.30.15:36760 - "GET /proxmox/endpoints HTTP/1.1" 429 Too Many Requests
```



And now the most important and the reason for this issue - I am not able to sync any data via the Full Sync job (or each at a time...)

The proxbox API seems to be okay - Log:
```
INFO:     192.168.30.15:44010 - "GET /auth/bootstrap-status HTTP/1.1" 200 OK
INFO:     192.168.30.15:44026 - "GET /netbox/endpoint HTTP/1.1" 200 OK
2026-05-16 13:04:06,788 - proxbox_api.ssrf - INFO - SSRF: auto-allowed IP 192.168.30.15 from NetBox endpoint configuration
INFO:     192.168.30.15:44028 - "PUT /netbox/endpoint/1 HTTP/1.1" 200 OK
INFO:     192.168.30.15:44060 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:44044 - "GET /proxmox/endpoints HTTP/1.1" 200 OK
INFO:     192.168.30.15:44078 - "GET /auth/bootstrap-status HTTP/1.1" 200 OK
2026-05-16 13:04:07,273 - proxbox_api.ssrf - INFO - SSRF: auto-allowed IP 192.168.20.10 from Proxmox endpoint configuration
INFO:     192.168.30.15:44088 - "GET /netbox/endpoint HTTP/1.1" 200 OK
2026-05-16 13:04:07,738 - proxbox_api.ssrf - INFO - SSRF: auto-allowed IP 192.168.30.15 from NetBox endpoint configuration
INFO:     192.168.30.15:44074 - "PUT /proxmox/endpoints/1 HTTP/1.1" 200 OK
INFO:     192.168.30.15:44090 - "PUT /netbox/endpoint/1 HTTP/1.1" 200 OK
INFO:     192.168.30.15:44104 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:44120 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:44136 - "GET / HTTP/1.1" 200 OK
proxbox [2026-05-16 13:04:07,997] [INFO] proxmox_core: INPUT is Pydantic Model ProxmoxSessionSchema
proxbox [2026-05-16 13:04:08,016] [INFO] proxmox_core: Using token to authenticate with Proxmox
proxbox [2026-05-16 13:04:08,016] [INFO] proxmox_core: Using domain 192.168.20.10 to authenticate with Proxmox
INFO:     192.168.30.15:44164 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:44172 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:44148 - "GET /proxmox/endpoints HTTP/1.1" 200 OK
INFO:     192.168.30.15:44098 - "GET /proxmox/cluster/status HTTP/1.1" 200 OK
INFO:     192.168.30.15:44190 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:44204 - "GET / HTTP/1.1" 200 OK
2026-05-16 13:04:08,809 - proxbox_api.ssrf - INFO - SSRF: auto-allowed IP 192.168.20.10 from Proxmox endpoint configuration
proxbox [2026-05-16 13:04:08,810] [INFO] proxmox_core: INPUT is Pydantic Model ProxmoxSessionSchema
proxbox [2026-05-16 13:04:08,811] [INFO] proxmox_core: Using token to authenticate with Proxmox
proxbox [2026-05-16 13:04:08,811] [INFO] proxmox_core: Using domain 192.168.20.10 to authenticate with Proxmox
INFO:     192.168.30.15:44180 - "PUT /proxmox/endpoints/1 HTTP/1.1" 200 OK
proxbox [2026-05-16 13:04:09,142] [INFO] proxmox_core: INPUT is Pydantic Model ProxmoxSessionSchema
proxbox [2026-05-16 13:04:09,142] [INFO] proxmox_core: Using token to authenticate with Proxmox
proxbox [2026-05-16 13:04:09,142] [INFO] proxmox_core: Using domain 192.168.20.10 to authenticate with Proxmox
INFO:     192.168.30.15:44224 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:44236 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:44208 - "GET /proxmox/nodes/ HTTP/1.1" 200 OK
INFO:     192.168.30.15:44214 - "GET /proxmox/version?source=database&domain=192.168.20.10 HTTP/1.1" 200 OK
INFO:     192.168.30.15:44248 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:44260 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:44266 - "GET /health HTTP/1.1" 200 OK
2026-05-16 13:04:09,746 - netbox_sdk.client - WARNING - HTTPS TLS certificate verification is disabled for https://192.168.30.15:443
INFO:     192.168.30.15:44284 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:44300 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:44278 - "GET /proxmox/endpoints HTTP/1.1" 200 OK
INFO:     192.168.30.15:44328 - "GET /proxmox/endpoints HTTP/1.1" 200 OK
2026-05-16 13:04:10,613 - proxbox_api.ssrf - INFO - SSRF: auto-allowed IP 192.168.20.10 from Proxmox endpoint configuration
2026-05-16 13:04:10,614 - netbox_sdk.client - INFO - api request starting
INFO:     192.168.30.15:44734 - "GET /proxmox/endpoints HTTP/1.1" 200 OK
proxbox [2026-05-16 13:04:10,926] [WARNING] structured_logging: [validation] No cluster status data provided
INFO:     192.168.30.15:44274 - "GET /dcim/devices/create/stream?use_guest_agent_interface_name=true&fetch_max_concurrency=2&ignore_ipv6_link_local_addresses=true&primary_ip_preference=ipv4&overwrite_device_role=true&overwrite_device_type=true&overwrite_device_tags=true&overwrite_device_status=true&overwrite_device_description=true&overwrite_device_custom_fields=true&overwrite_vm_role=true&overwrite_vm_type=true&overwrite_vm_tags=true&overwrite_vm_description=true&overwrite_vm_custom_fields=true&overwrite_cluster_tags=true&overwrite_cluster_description=true&overwrite_cluster_custom_fields=true&overwrite_node_interface_tags=true&overwrite_node_interface_custom_fields=true&overwrite_storage_tags=true&overwrite_vm_interface_tags=true&overwrite_vm_interface_custom_fields=true&overwrite_ip_status=true&overwrite_ip_tags=true&overwrite_ip_custom_fields=true&overwrite_ip_address_dns_name=true&proxmox_endpoint_ids=4 HTTP/1.1" 200 OK
INFO:     192.168.30.15:44314 - "PUT /proxmox/endpoints/1 HTTP/1.1" 200 OK
INFO:     192.168.30.15:44770 - "GET /health HTTP/1.1" 200 OK
2026-05-16 13:04:11,691 - proxbox_api.ssrf - INFO - SSRF: auto-allowed IP 192.168.20.10 from Proxmox endpoint configuration
proxbox [2026-05-16 13:04:11,700] [INFO] proxmox_core: INPUT is Pydantic Model ProxmoxSessionSchema
proxbox [2026-05-16 13:04:11,701] [INFO] proxmox_core: Using token to authenticate with Proxmox
proxbox [2026-05-16 13:04:11,701] [INFO] proxmox_core: Using domain 192.168.20.10 to authenticate with Proxmox
INFO:     192.168.30.15:44750 - "PUT /proxmox/endpoints/1 HTTP/1.1" 200 OK
INFO:     192.168.30.15:44764 - "PUT /proxmox/endpoints/1 HTTP/1.1" 200 OK
proxbox [2026-05-16 13:04:11,715] [INFO] storages: No Proxmox sessions available for storage sync
INFO:     192.168.30.15:44796 - "GET /virtualization/virtual-machines/storage/create/stream?use_guest_agent_interface_name=true&fetch_max_concurrency=2&ignore_ipv6_link_local_addresses=true&primary_ip_preference=ipv4&overwrite_device_role=true&overwrite_device_type=true&overwrite_device_tags=true&overwrite_device_status=true&overwrite_device_description=true&overwrite_device_custom_fields=true&overwrite_vm_role=true&overwrite_vm_type=true&overwrite_vm_tags=true&overwrite_vm_description=true&overwrite_vm_custom_fields=true&overwrite_cluster_tags=true&overwrite_cluster_description=true&overwrite_cluster_custom_fields=true&overwrite_node_interface_tags=true&overwrite_node_interface_custom_fields=true&overwrite_storage_tags=true&overwrite_vm_interface_tags=true&overwrite_vm_interface_custom_fields=true&overwrite_ip_status=true&overwrite_ip_tags=true&overwrite_ip_custom_fields=true&overwrite_ip_address_dns_name=true&proxmox_endpoint_ids=4 HTTP/1.1" 200 OK
INFO:     192.168.30.15:44784 - "GET /proxmox/version?source=database&domain=192.168.20.10 HTTP/1.1" 200 OK
proxbox [2026-05-16 13:04:12,237] [INFO] proxmox_core: INPUT is Pydantic Model ProxmoxSessionSchema
proxbox [2026-05-16 13:04:12,238] [INFO] proxmox_core: Using token to authenticate with Proxmox
proxbox [2026-05-16 13:04:12,238] [INFO] proxmox_core: Using domain 192.168.20.10 to authenticate with Proxmox
INFO:     192.168.30.15:44818 - "GET /health HTTP/1.1" 200 OK
INFO:     192.168.30.15:44830 - "GET / HTTP/1.1" 200 OK
proxbox [2026-05-16 13:04:12,240] [INFO] proxmox_core: INPUT is Pydantic Model ProxmoxSessionSchema
proxbox [2026-05-16 13:04:12,240] [INFO] proxmox_core: Using token to authenticate with Proxmox
proxbox [2026-05-16 13:04:12,240] [INFO] proxmox_core: Using domain 192.168.20.10 to authenticate with Proxmox
INFO:     192.168.30.15:44836 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:44800 - "GET /proxmox/version?source=database&domain=192.168.20.10 HTTP/1.1" 200 OK
INFO:     192.168.30.15:44814 - "GET /proxmox/version?source=database&domain=192.168.20.10 HTTP/1.1" 200 OK
2026-05-16 13:04:12,794 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:12,795 - netbox_sdk.client - INFO - api request starting
proxbox [2026-05-16 13:04:13,124] [INFO] proxmox_core: INPUT is Pydantic Model ProxmoxSessionSchema
proxbox [2026-05-16 13:04:13,124] [INFO] proxmox_core: Using token to authenticate with Proxmox
proxbox [2026-05-16 13:04:13,124] [INFO] proxmox_core: Using domain 192.168.20.10 to authenticate with Proxmox
INFO:     192.168.30.15:44848 - "GET /proxmox/endpoints HTTP/1.1" 200 OK
2026-05-16 13:04:13,414 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:13,414 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:13,415 - netbox_sdk.client - INFO - api request starting
INFO:     192.168.30.15:44842 - "GET /proxmox/sessions?source=database&domain=192.168.20.10 HTTP/1.1" 200 OK
2026-05-16 13:04:13,442 - proxbox_api.ssrf - INFO - SSRF: auto-allowed IP 192.168.20.10 from Proxmox endpoint configuration
INFO:     192.168.30.15:44850 - "PUT /proxmox/endpoints/1 HTTP/1.1" 200 OK
2026-05-16 13:04:13,702 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:13,702 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:13,703 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:13,948 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:13,948 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:13,949 - netbox_sdk.client - INFO - api request starting
proxbox [2026-05-16 13:04:13,971] [INFO] proxmox_core: INPUT is Pydantic Model ProxmoxSessionSchema
proxbox [2026-05-16 13:04:13,971] [INFO] proxmox_core: Using token to authenticate with Proxmox
proxbox [2026-05-16 13:04:13,972] [INFO] proxmox_core: Using domain 192.168.20.10 to authenticate with Proxmox
INFO:     192.168.30.15:44868 - "GET /proxmox/endpoints HTTP/1.1" 200 OK
2026-05-16 13:04:14,227 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:14,228 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:14,228 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:14,242 - proxbox_api.ssrf - INFO - SSRF: auto-allowed IP 192.168.20.10 from Proxmox endpoint configuration
INFO:     192.168.30.15:44866 - "GET /proxmox/version?source=database&domain=192.168.20.10 HTTP/1.1" 200 OK
INFO:     192.168.30.15:44884 - "PUT /proxmox/endpoints/1 HTTP/1.1" 200 OK
2026-05-16 13:04:14,519 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:14,519 - netbox_sdk.client - INFO - api request starting
proxbox [2026-05-16 13:04:14,527] [INFO] proxmox_core: INPUT is Pydantic Model ProxmoxSessionSchema
proxbox [2026-05-16 13:04:14,527] [INFO] proxmox_core: Using token to authenticate with Proxmox
proxbox [2026-05-16 13:04:14,527] [INFO] proxmox_core: Using domain 192.168.20.10 to authenticate with Proxmox
INFO:     192.168.30.15:44886 - "GET /proxmox/version?source=database&domain=192.168.20.10 HTTP/1.1" 200 OK
2026-05-16 13:04:14,795 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:14,796 - netbox_sdk.client - INFO - api request starting
proxbox [2026-05-16 13:04:14,805] [INFO] proxmox_core: INPUT is Pydantic Model ProxmoxSessionSchema
proxbox [2026-05-16 13:04:14,806] [INFO] proxmox_core: Using token to authenticate with Proxmox
proxbox [2026-05-16 13:04:14,806] [INFO] proxmox_core: Using domain 192.168.20.10 to authenticate with Proxmox
INFO:     192.168.30.15:44892 - "GET /proxmox/sessions?source=database&domain=192.168.20.10 HTTP/1.1" 200 OK
2026-05-16 13:04:14,857 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:14,857 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:14,949 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:14,950 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:15,044 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:15,044 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:15,087 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:15,088 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:15,144 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:15,144 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:15,184 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:15,185 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:15,237 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:15,238 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:15,281 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:15,282 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:15,335 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:15,335 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:15,394 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:15,397 - netbox_sdk.client - INFO - api request starting
INFO:     192.168.30.15:44904 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:44918 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:44930 - "GET /auth/bootstrap-status HTTP/1.1" 200 OK
INFO:     192.168.30.15:44940 - "GET /auth/bootstrap-status HTTP/1.1" 200 OK
2026-05-16 13:04:15,753 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:15,754 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:16,033 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:16,034 - netbox_sdk.client - INFO - api request starting
INFO:     192.168.30.15:44966 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:44950 - "GET /netbox/endpoint HTTP/1.1" 200 OK
INFO:     192.168.30.15:44964 - "GET /netbox/endpoint HTTP/1.1" 200 OK
2026-05-16 13:04:16,283 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:16,284 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:16,517 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:16,517 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:16,533 - proxbox_api.ssrf - INFO - SSRF: auto-allowed IP 192.168.30.15 from NetBox endpoint configuration
INFO:     192.168.30.15:45006 - "GET /auth/bootstrap-status HTTP/1.1" 200 OK
INFO:     192.168.30.15:44982 - "PUT /netbox/endpoint/1 HTTP/1.1" 200 OK
INFO:     192.168.30.15:44994 - "PUT /netbox/endpoint/1 HTTP/1.1" 200 OK
2026-05-16 13:04:16,779 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:16,779 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:17,235 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:17,236 - netbox_sdk.client - INFO - api request starting
INFO:     192.168.30.15:45022 - "GET /netbox/endpoint HTTP/1.1" 200 OK
INFO:     192.168.30.15:45026 - "GET /proxmox/endpoints HTTP/1.1" 200 OK
2026-05-16 13:04:17,703 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:17,703 - netbox_sdk.client - INFO - api request starting
INFO:     192.168.30.15:45030 - "GET /proxmox/endpoints HTTP/1.1" 200 OK
2026-05-16 13:04:17,947 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:17,948 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:17,986 - proxbox_api.ssrf - INFO - SSRF: auto-allowed IP 192.168.30.15 from NetBox endpoint configuration
2026-05-16 13:04:17,988 - proxbox_api.ssrf - INFO - SSRF: auto-allowed IP 192.168.20.10 from Proxmox endpoint configuration
INFO:     192.168.30.15:45046 - "PUT /netbox/endpoint/1 HTTP/1.1" 200 OK
2026-05-16 13:04:18,000 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:18,000 - netbox_sdk.client - INFO - api request starting
INFO:     192.168.30.15:45048 - "PUT /proxmox/endpoints/1 HTTP/1.1" 200 OK
INFO:     192.168.30.15:45070 - "PUT /proxmox/endpoints/1 HTTP/1.1" 200 OK
2026-05-16 13:04:18,503 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:18,504 - netbox_sdk.client - INFO - api request starting
INFO:     192.168.30.15:45082 - "GET / HTTP/1.1" 200 OK
INFO:     192.168.30.15:45094 - "GET / HTTP/1.1" 200 OK
proxbox [2026-05-16 13:04:18,528] [INFO] proxmox_core: INPUT is Pydantic Model ProxmoxSessionSchema
proxbox [2026-05-16 13:04:18,529] [INFO] proxmox_core: Using token to authenticate with Proxmox
proxbox [2026-05-16 13:04:18,529] [INFO] proxmox_core: Using domain 192.168.20.10 to authenticate with Proxmox
proxbox [2026-05-16 13:04:18,531] [INFO] proxmox_core: INPUT is Pydantic Model ProxmoxSessionSchema
proxbox [2026-05-16 13:04:18,531] [INFO] proxmox_core: Using token to authenticate with Proxmox
proxbox [2026-05-16 13:04:18,531] [INFO] proxmox_core: Using domain 192.168.20.10 to authenticate with Proxmox
INFO:     192.168.30.15:45108 - "GET /auth/bootstrap-status HTTP/1.1" 200 OK
INFO:     192.168.30.15:45080 - "GET /proxmox/version?source=database&domain=192.168.20.10 HTTP/1.1" 200 OK
INFO:     192.168.30.15:45078 - "GET /proxmox/version?source=database&domain=192.168.20.10 HTTP/1.1" 200 OK
2026-05-16 13:04:18,828 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:18,829 - netbox_sdk.client - INFO - api request starting
INFO:     192.168.30.15:45120 - "GET /netbox/endpoint HTTP/1.1" 200 OK
2026-05-16 13:04:19,065 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:19,066 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:19,075 - proxbox_api.ssrf - INFO - SSRF: auto-allowed IP 192.168.30.15 from NetBox endpoint configuration
INFO:     192.168.30.15:45132 - "PUT /netbox/endpoint/1 HTTP/1.1" 200 OK
2026-05-16 13:04:19,309 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:19,309 - netbox_sdk.client - INFO - api request starting
INFO:     192.168.30.15:45142 - "GET /proxmox/endpoints HTTP/1.1" 200 OK
2026-05-16 13:04:19,559 - proxbox_api.ssrf - INFO - SSRF: auto-allowed IP 192.168.20.10 from Proxmox endpoint configuration
INFO:     192.168.30.15:45146 - "PUT /proxmox/endpoints/1 HTTP/1.1" 200 OK
2026-05-16 13:04:19,792 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:19,793 - netbox_sdk.client - INFO - api request starting
proxbox [2026-05-16 13:04:19,801] [INFO] proxmox_core: INPUT is Pydantic Model ProxmoxSessionSchema
proxbox [2026-05-16 13:04:19,801] [INFO] proxmox_core: Using token to authenticate with Proxmox
proxbox [2026-05-16 13:04:19,801] [INFO] proxmox_core: Using domain 192.168.20.10 to authenticate with Proxmox
INFO:     192.168.30.15:45154 - "GET /proxmox/version?source=database&domain=192.168.20.10 HTTP/1.1" 200 OK
2026-05-16 13:04:19,837 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:19,837 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:19,864 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:19,865 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:19,906 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:19,906 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:19,933 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:19,934 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:19,970 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:19,971 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,002 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,003 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,040 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,040 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,067 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,068 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,109 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,109 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,137 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,137 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,175 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,175 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,206 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,207 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,244 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,244 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,271 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,272 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,313 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,314 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,339 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,340 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,374 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,375 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,400 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,401 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,445 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,446 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,474 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,475 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,512 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,513 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,544 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,545 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,580 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,580 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,606 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,606 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,646 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,646 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,672 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,673 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,710 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,711 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,741 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,741 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,776 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,776 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,802 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,803 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,844 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,844 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,845 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,881 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,882 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,882 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,917 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,917 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,918 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,952 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,952 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:20,984 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:20,985 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:21,020 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:21,021 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:21,046 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:21,047 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:21,087 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:21,087 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:21,113 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:21,114 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:21,149 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:21,149 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:21,180 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:21,181 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:21,216 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:21,216 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:21,242 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:21,243 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:21,282 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:21,282 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:21,308 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:21,309 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:21,344 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:21,345 - netbox_sdk.client - INFO - api request starting
INFO:     192.168.30.15:44832 - "GET /virtualization/virtual-machines/create/stream?use_guest_agent_interface_name=true&fetch_max_concurrency=2&ignore_ipv6_link_local_addresses=true&primary_ip_preference=ipv4&overwrite_device_role=true&overwrite_device_type=true&overwrite_device_tags=true&overwrite_device_status=true&overwrite_device_description=true&overwrite_device_custom_fields=true&overwrite_vm_role=true&overwrite_vm_type=true&overwrite_vm_tags=true&overwrite_vm_description=true&overwrite_vm_custom_fields=true&overwrite_cluster_tags=true&overwrite_cluster_description=true&overwrite_cluster_custom_fields=true&overwrite_node_interface_tags=true&overwrite_node_interface_custom_fields=true&overwrite_storage_tags=true&overwrite_vm_interface_tags=true&overwrite_vm_interface_custom_fields=true&overwrite_ip_status=true&overwrite_ip_tags=true&overwrite_ip_custom_fields=true&overwrite_ip_address_dns_name=true&proxmox_endpoint_ids=4&sync_vm_network=false HTTP/1.1" 200 OK
2026-05-16 13:04:21,353 - netbox_sdk.client - INFO - api request completed
2026-05-16 13:04:21,353 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:21,379 - netbox_sdk.client - INFO - api request completed
proxbox [2026-05-16 13:04:21,380] [INFO] sync_vm: VM sync summary: total=0 ok=0 failed=0
INFO:     192.168.30.15:60488 - "GET /health HTTP/1.1" 200 OK
2026-05-16 13:04:21,621 - netbox_sdk.client - WARNING - HTTPS TLS certificate verification is disabled for https://192.168.30.15:443
2026-05-16 13:04:21,669 - netbox_sdk.client - INFO - api request starting
2026-05-16 13:04:21,670 - netbox_sdk.client - INFO - api request starting
INFO:     192.168.30.15:60498 - "GET /virtualization/virtual-machines/task-history/create/stream?use_guest_agent_interface_name=true&fetch_max_concurrency=2&ignore_ipv6_link_local_addresses=true&primary_ip_preference=ipv4&overwrite_device_role=true&overwrite_device_type=true&overwrite_device_tags=true&overwrite_device_status=true&overwrite_device_description=true&overwrite_device_custom_fields=true&overwrite_vm_role=true&overwrite_vm_type=true&overwrite_vm_tags=true&overwrite_vm_description=true&overwrite_vm_custom_fields=true&overwrite_cluster_tags=true&overwrite_cluster_description=true&overwrite_cluster_custom_fields=true&overwrite_node_interface_tags=true&overwrite_node_interface_custom_fields=true&overwrite_storage_tags=true&overwrite_vm_interface_tags=true&overwrite_vm_interface_custom_fields=true&overwrite_ip_status=true&overwrite_ip_tags=true&overwrite_ip_custom_fields=true&overwrite_ip_address_dns_name=true&proxmox_endpoint_ids=4 HTTP/1.1" 200 OK
2026-05-16 13:04:21,807 - netbox_sdk.client - INFO - api request completed
proxbox [2026-05-16 13:04:21,808] [INFO] task_history: No VMs found with cf_proxmox_vm_id for task history sync
INFO:     192.168.30.15:60508 - "GET /health HTTP/1.1" 200 OK
proxbox [2026-05-16 13:04:22,052] [INFO] virtual_disks: Starting virtual disks sync for existing VMs
2026-05-16 13:04:22,052 - netbox_sdk.client - INFO - api request starting
proxbox [2026-05-16 13:04:22,052] [INFO] virtual_disks: No VMs found with cf_proxmox_vm_id set
INFO:     192.168.30.15:60524 - "GET /virtualization/virtual-machines/virtual-disks/create/stream?use_guest_agent_interface_name=true&fetch_max_concurrency=2&ignore_ipv6_link_local_addresses=true&primary_ip_preference=ipv4&overwrite_device_role=true&overwrite_device_type=true&overwrite_device_tags=true&overwrite_device_status=true&overwrite_device_description=true&overwrite_device_custom_fields=true&overwrite_vm_role=true&overwrite_vm_type=true&overwrite_vm_tags=true&overwrite_vm_description=true&overwrite_vm_custom_fields=true&overwrite_cluster_tags=true&overwrite_cluster_description=true&overwrite_cluster_custom_fields=true&overwrite_node_interface_tags=true&overwrite_node_interface_custom_fields=true&overwrite_storage_tags=true&overwrite_vm_interface_tags=true&overwrite_vm_interface_custom_fields=true&overwrite_ip_status=true&overwrite_ip_tags=true&overwrite_ip_custom_fields=true&overwrite_ip_address_dns_name=true&proxmox_endpoint_ids=4 HTTP/1.1" 200 OK
INFO:     192.168.30.15:60540 - "GET /health HTTP/1.1" 200 OK
2026-05-16 13:04:22,296 - netbox_sdk.client - INFO - api request starting
INFO:     192.168.30.15:60546 - "GET /virtualization/virtual-machines/backups/all/create/stream?use_guest_agent_interface_name=true&fetch_max_concurrency=2&ignore_ipv6_link_local_addresses=true&primary_ip_preference=ipv4&overwrite_device_role=true&overwrite_device_type=true&overwrite_device_tags=true&overwrite_device_status=true&overwrite_device_description=true&overwrite_device_custom_fields=true&overwrite_vm_role=true&overwrite_vm_type=true&overwrite_vm_tags=true&overwrite_vm_description=true&overwrite_vm_custom_fields=true&overwrite_cluster_tags=true&overwrite_cluster_description=true&overwrite_cluster_custom_fields=true&overwrite_node_interface_tags=true&overwrite_node_interface_custom_fields=true&overwrite_storage_tags=true&overwrite_vm_interface_tags=true&overwrite_vm_interface_custom_fields=true&overwrite_ip_status=true&overwrite_ip_tags=true&overwrite_ip_custom_fields=true&overwrite_ip_address_dns_name=true&proxmox_endpoint_ids=4&delete_nonexistent_backup=true HTTP/1.1" 200 OK
2026-05-16 13:04:22,395 - netbox_sdk.client - INFO - api request completed
proxbox [2026-05-16 13:04:22,396] [DEBUG] exception: ProxboxException: No backups found to process
INFO:     192.168.30.15:58638 - "GET /health HTTP/1.1" 200 OK
proxbox [2026-05-16 13:04:30,642] [DEBUG] exception: ProxboxException: No backups found to process
INFO:     192.168.30.15:58646 - "GET /virtualization/virtual-machines/backups/all/create/stream?use_guest_agent_interface_name=true&fetch_max_concurrency=2&ignore_ipv6_link_local_addresses=true&primary_ip_preference=ipv4&overwrite_device_role=true&overwrite_device_type=true&overwrite_device_tags=true&overwrite_device_status=true&overwrite_device_description=true&overwrite_device_custom_fields=true&overwrite_vm_role=true&overwrite_vm_type=true&overwrite_vm_tags=true&overwrite_vm_description=true&overwrite_vm_custom_fields=true&overwrite_cluster_tags=true&overwrite_cluster_description=true&overwrite_cluster_custom_fields=true&overwrite_node_interface_tags=true&overwrite_node_interface_custom_fields=true&overwrite_storage_tags=true&overwrite_vm_interface_tags=true&overwrite_vm_interface_custom_fields=true&overwrite_ip_status=true&overwrite_ip_tags=true&overwrite_ip_custom_fields=true&overwrite_ip_address_dns_name=true&proxmox_endpoint_ids=4&delete_nonexistent_backup=true HTTP/1.1" 200 OK
INFO:     192.168.30.15:58662 - "GET /health HTTP/1.1" 200 OK
proxbox [2026-05-16 13:04:38,914] [DEBUG] exception: ProxboxException: No backups found to process
INFO:     192.168.30.15:58668 - "GET /virtualization/virtual-machines/backups/all/create/stream?use_guest_agent_interface_name=true&fetch_max_concurrency=2&ignore_ipv6_link_local_addresses=true&primary_ip_preference=ipv4&overwrite_device_role=true&overwrite_device_type=true&overwrite_device_tags=true&overwrite_device_status=true&overwrite_device_description=true&overwrite_device_custom_fields=true&overwrite_vm_role=true&overwrite_vm_type=true&overwrite_vm_tags=true&overwrite_vm_description=true&overwrite_vm_custom_fields=true&overwrite_cluster_tags=true&overwrite_cluster_description=true&overwrite_cluster_custom_fields=true&overwrite_node_interface_tags=true&overwrite_node_interface_custom_fields=true&overwrite_storage_tags=true&overwrite_vm_interface_tags=true&overwrite_vm_interface_custom_fields=true&overwrite_ip_status=true&overwrite_ip_tags=true&overwrite_ip_custom_fields=true&overwrite_ip_address_dns_name=true&proxmox_endpoint_ids=4&delete_nonexistent_backup=true HTTP/1.1" 200 OK


```

but it seems that the plugins seems to except backips for vms? - I have a backup job configured and tehre are backups on each vm (vzdumps) - but it doesn't seem to find these and (for me) it should not be a reason to fail - maybe I undestand the error wrong :/
I already looked online but I cannot find any documentation about a rate limiting setting for PVE...
Further I already set all values way up in the settings and set the retries for proxmox to 10 with 2s delay...

This is the Logoutput for the full sync netbox job error log -  I also enabled Debug mode but there are no debug messages...
```

2026-05-16 11:04:06.266 	Info 	Starting Proxbox sync stages: devices, storage, virtual-machines, task-history, vm-disks, vm-backups, vm-snapshots, network-interfaces, vm-interfaces, ip-addresses, replications, backup-routines
2026-05-16 11:04:06.266 	Info 	Proxmox endpoints: [4]
2026-05-16 11:04:06.295 	Info 	Preflight: API key verified — %s
2026-05-16 11:04:06.793 	Info 	Preflight: synced NetBox endpoint '%s' to proxbox-api backend
2026-05-16 11:04:07.740 	Info 	Preflight: synced Proxmox endpoint '%s' to proxbox-api backend
2026-05-16 11:04:07.742 	Info 	Syncing cluster/nodes for endpoint 4
2026-05-16 11:04:09.197 	Info 	Cluster/node sync for endpoint 4: 0 cluster(s) created, 1 updated, 0 node(s) created, 0 updated
2026-05-16 11:04:09.197 	Info 	Running SSE sync for Proxmox endpoint %s
2026-05-16 11:04:09.207 	Info 	Starting stage: devices (dcim/devices/create/stream)
2026-05-16 11:04:09.207 	Info 	Checking backend readiness for stage 'devices'...
2026-05-16 11:04:10.951 	Info 	stepstatus started · message Starting devices synchronization.
2026-05-16 11:04:10.963 	Info 	Stage completed: devices (dcim/devices/create/stream) HTTP 200 in 1s
2026-05-16 11:04:10.963 	Info 	Starting stage: storage (virtualization/virtual-machines/storage/create/stream)
2026-05-16 11:04:10.963 	Info 	Checking backend readiness for stage 'storage'...
2026-05-16 11:04:11.750 	Info 	Stage completed: storage (virtualization/virtual-machines/storage/create/stream) HTTP 200 in 0s
2026-05-16 11:04:11.750 	Info 	Starting stage: virtual-machines (virtualization/virtual-machines/create/stream)
2026-05-16 11:04:11.750 	Info 	Checking backend readiness for stage 'virtual-machines'...
2026-05-16 11:04:21.347 	Info 	stepstatus started · message Starting virtual machines synchronization.
2026-05-16 11:04:21.384 	Info 	Stage completed: virtual-machines (virtualization/virtual-machines/create/stream) HTTP 200 in 9s
2026-05-16 11:04:21.384 	Info 	Starting stage: task-history (virtualization/virtual-machines/task-history/create/stream)
2026-05-16 11:04:21.384 	Info 	Checking backend readiness for stage 'task-history'...
2026-05-16 11:04:21.812 	Info 	Stage completed: task-history (virtualization/virtual-machines/task-history/create/stream) HTTP 200 in 0s
2026-05-16 11:04:21.812 	Info 	Starting stage: vm-disks (virtualization/virtual-machines/virtual-disks/create/stream)
2026-05-16 11:04:21.812 	Info 	Checking backend readiness for stage 'vm-disks'...
2026-05-16 11:04:22.057 	Info 	Stage completed: vm-disks (virtualization/virtual-machines/virtual-disks/create/stream) HTTP 200 in 0s
2026-05-16 11:04:22.057 	Info 	Starting stage: vm-backups (virtualization/virtual-machines/backups/all/create/stream)
2026-05-16 11:04:22.057 	Info 	Checking backend readiness for stage 'vm-backups'...
2026-05-16 11:04:22.396 	Info 	errorstatus failed · error No backups found to process · detail No backups found to process
2026-05-16 11:04:22.400 	Warning 	Stage vm-backups failed (HTTP 503): No backups found to process -- retrying in 8s (attempt 1/2)
2026-05-16 11:04:30.402 	Info 	Checking backend readiness for stage 'vm-backups'...
2026-05-16 11:04:30.643 	Info 	stepstatus started · message Starting backups synchronization.
2026-05-16 11:04:30.646 	Info 	errorstatus failed · error No backups found to process · detail No backups found to process
2026-05-16 11:04:30.649 	Warning 	Stage vm-backups failed (HTTP 503): No backups found to process -- retrying in 8s (attempt 2/2)
2026-05-16 11:04:38.651 	Info 	Checking backend readiness for stage 'vm-backups'...
2026-05-16 11:04:38.917 	Info 	stepstatus started · message Starting backups synchronization.
2026-05-16 11:04:38.922 	Info 	errorstatus failed · error No backups found to process · detail No backups found to process
2026-05-16 11:04:38.925 	Error 	Stage vm-backups failed (HTTP 503): No backups found to process
2026-05-16 11:04:38.927 	Error 	

Traceback (most recent call last):
  File "netbox/netbox/jobs.py", line 110, in handle
    cls(job).run(*args, **kwargs)
    ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "netbox/netbox-proxbox/netbox_proxbox/jobs.py", line 491, in run
    stages_out = _run_all_stages_sync(self, stages, params, run_started)
  File "netbox/netbox-proxbox/netbox_proxbox/jobs.py", line 120, in _run_all_stages_sync
    return sync_stages._run_all_stages_sync(*args, **kwargs)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "netbox/netbox-proxbox/netbox_proxbox/sync_stages.py", line 409, in _run_all_stages_sync
    payload, stage_runtime = _execute_stage_sync(
                             ~~~~~~~~~~~~~~~~~~~^
        job, st, stream_path, query_params, on_frame, fastapi_endpoint_id
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "netbox/netbox-proxbox/netbox_proxbox/sync_stages.py", line 313, in _execute_stage_sync
    raise RuntimeError(user_detail)
RuntimeError: Stage 'vm-backups' failed (HTTP 503): No backups found to process

```

Thank you for your help :) - If I can be of assistance, please tell what I need to do.

--- a/netbox_proxbox/sync_stages.py
+++ b/netbox_proxbox/sync_stages.py
@@ -44,6 +44,17 @@ _HEARTBEAT_SECONDS = 20.0
 _STAGE_RETRY_MAX = 2
 _STAGE_RETRY_DELAY = 8.0
 
+# Stages that are supplementary/optional: a failure logs a warning and the sync
+# continues.  Required stages (devices, VMs, storage, interfaces, IPs) are NOT
+# in this set and still abort the run on failure.
+_SKIPPABLE_STAGES: frozenset[str] = frozenset(
+    {
+        SyncTypeChoices.VIRTUAL_MACHINES_BACKUPS,   # "vm-backups"
+        SyncTypeChoices.VIRTUAL_MACHINES_SNAPSHOTS, # "vm-snapshots"
+        SyncTypeChoices.TASK_HISTORY,               # "task-history"
+    }
+)
+
 
 async def _run_batch_selected_sync(
     self: "ProxboxSyncJob",
@@ -410,9 +421,18 @@ def _run_all_stages_sync(
             stage_paths = _sync_stream_paths_for_stage(st, target_vm_ids)
 
             for stream_path in stage_paths:
-                payload, stage_runtime = _execute_stage_sync(
-                    job, st, stream_path, query_params, on_frame, fastapi_endpoint_id
-                )
+                try:
+                    payload, stage_runtime = _execute_stage_sync(
+                        job, st, stream_path, query_params, on_frame, fastapi_endpoint_id
+                    )
+                except RuntimeError as exc:
+                    if st in _SKIPPABLE_STAGES:
+                        job.logger.warning(
+                            "Optional stage '%s' failed and was skipped: %s", st, exc
+                        )
+                        job.job.save(update_fields=["log_entries"])
+                        continue
+                    raise
                 response = payload.get("response") or {}
                 stages_out.append(
                     {

